packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -85,26 +95,56 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
     delete versionsByLocale[version.locale];
 
     // For each locale, get the ones with the same status
+    // There will not be a draft and a version counterpart if the content
+    // type does not have draft and publish
+
+    const keysToKeep = [...AVAILABLE_LOCALES_FIELDS, ...validatableFields];
+    const mappingResult = await async.map(
+      Object.values(versionsByLocale),
+      async (localeVersions: DocumentVersion[]) => {
+        const model = strapi.getModel(uid);
+
+        const mappedLocaleVersions: DocumentVersion[] = await async.map(
+          localeVersions,
+          async (localeVersion: DocumentVersion) =>
```

Comment:
It looks like this method could also be declared just once, outside all of these nested maps unless I'm missing something which is very possible 😆

Reply:
thanks!